generators/client/templates/src/components/authentication/AuthenticatedRoute.js

```diff
@@ -2,21 +2,23 @@ import React from "react";
 import { Redirect, Route } from "react-router";
 
 const AuthenticationCheck = ({ component: Component, user }) => {
+  if (user === undefined) {
+    return <div>Loading...</div>
+  }
   if (user !== null) {
-    return <Component />;
+    return <Component user={user} />;
```

Comment:
I don't think we want to pass the `user` prop to every component. The context helps us with that. What component is trying to access the `user` as a prop, and can it use the context instead?

Reply:
@dpickett Can you help guide me toward a better pattern when using AuthenticatedRoute?

Would we do something like: 

```
<AuthenticatedRoute user={currentUser}> 
     <ComponentIWantToRender  user={currentUser} />
</AuthenticatedRoute> 
```